openzl-crypto/src/poseidon/constants.rs

```diff
@@ -0,0 +1,307 @@
+//! Poseidon Permutation Round Numbers
+
+use crate::poseidon::Specification;
+
+#[cfg(feature = "std")]
+use {openzl_util::num::Ceil, security::SecurityCondition};
+
+/// Poseidon Security Parameters
+///
+/// See the [`Specification`](super::Specification) for more on how these constants are part of the
+/// Poseidon Permutation algorithm.
+///
+/// # Safety
+///
+/// The constants that are specified to be "secure" in this `struct` have only been guaranteed to
+/// work for BLS12-381.
+#[derive(Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Constants {
+    /// Width of the Internal Poseidon State
+    pub width: usize,
+
+    /// Number of Full Rounds
+    pub full_rounds: usize,
+
+    /// Number of Partial Rounds
+    pub partial_rounds: usize,
+}
+
+impl Constants {
+    /// Poseidon Prime Field Modulus Bit Count
+    ///
+    /// This constant is denoted `n` in the Poseidon paper where `n = ceil(log2(p))`.
+    pub const MODULUS_BITS: usize = 255;
+
+    /// Security Level in Bits
+    ///
+    /// This constant is denoted `M` in the Poseidon paper.
+    pub const SECURITY_LEVEL: usize = 128;
```

Comment:
I agree, it would make more sense for them to be general. Though in light of \[this\](https://github.com/openzklib/openzl/blob/ef2dd89d954ad446b6ddc2f67815ba5f27df9ee9/openzl-crypto/src/poseidon/constants.rs#L13-L16) note it seems that maybe there's a restriction on which curves this algorithm applies to

Reply:
Moved to #44.